activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -80,12 +80,26 @@ def without(*elements)
   #   [{ id: 1, name: "David" }, { id: 2, name: "Rafael" }].pluck(:id, :name)
   #     => [[1, "David"], [2, "Rafael"]]
   def pluck(*keys)
-    if keys.many?
-      map { |element| keys.map { |key| element[key] } }
-    else
-      map { |element| element[keys.first] }
+    entries = map { |element| keys.map { |key| pluck_single(element, key) } }
+
+    return entries if keys.many?
+
+    entries.flatten!
+  end
+
+  # :nodoc:
+  def pluck_single(element, key)
+    if element.respond_to?(:has_attribute?) # AR
```

Comment:
Yeah `Enumerable#pluck` would implement the fallback branch in the current code and each class would have its own implementation.